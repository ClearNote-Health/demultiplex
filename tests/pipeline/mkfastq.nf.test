nextflow_pipeline {

    name "Test Workflow main.nf - mkfastq"
    script "main.nf"
    profile "test_mkfastq"
    tag "mkfastq"
    tag "pipeline"

    test("MKFASTQ") {

        when {
            params {
                outdir        = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert workflow.trace.succeeded().size() == 12 },
                // How many directories were produced?
                {assert path("${outputDir}").list().size() == 6},
                // How many files were produced?
                {assert path("$outputDir/cellranger-tiny-bcl-simple/").list().size() == 2},
                {assert path("$outputDir/multiqc/").list().size() == 3},
                {assert path("$outputDir/pipeline_info/").list().size() == 5},
                // File assertions: Files that should be exactly the same with every pipeline execution.
                {assert snapshot(
                    //Check output files
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/test_sample_S1_L001_1.fastp.fastq.gz"),
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/test_sample_S1_L001_1.fastp.fastq.gz.md5"),
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/test_sample_S1_L001_2.fastp.fastq.gz"),
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/test_sample_S1_L001_2.fastp.fastq.gz.md5"),
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/test_sample_S1_L001_I1_001.fastp.fastq.gz.md5"),
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/test_sample_S1_L001_I1_001.fastq.gz"),
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/test_sample_S1_L001_R1_001.fastq.gz"),
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/test_sample_S1_L001_R2_001.fastq.gz"),
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/undetermined/Undetermined_S0_L001_I1_001.fastq.gz"),
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/undetermined/Undetermined_S0_L001_R1_001.fastq.gz"),
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/undetermined/Undetermined_S0_L001_R2_001.fastq.gz"),
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/test_sample_S1_L001_I1_001.fastp.html").exists(),
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/test_sample_S1_L001.fastp.html").exists(),
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/test_sample_S1_L001.fastp.json"),
                    path("${outputDir}/cellranger-tiny-bcl-simple/L001/test_sample_S1_L001_I1_001.fastp.json"),
                    path("${outputDir}/cellranger-tiny-bcl-simple/InterOp/IndexMetricsOut.bin")
                ).match("file_assertions_mkfastq")}
            )
        }
    }
}
