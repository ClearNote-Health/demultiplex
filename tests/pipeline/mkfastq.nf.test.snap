{
    "software_versions": {
        "content": [
            "{CELLRANGER_MKFASTQ={cellranger=8.0.0}, FASTP={fastp=0.23.4}, MD5SUM={md5sum=8.3}, UNTAR_FLOWCELL={untar=1.34}, Workflow={nf-core/demultiplex=v1.5.2}}"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-15T17:36:35.551329"
    },
    "file_assertions_mkfastq": {
        "content": [
            "test_sample_S1_L001_1.fastp.fastq.gz:md5,b348491a3b1afa096fde514dbdf55215",
            "test_sample_S1_L001_1.fastp.fastq.gz.md5:md5,43d7882f3aaae47b56df56da6c802236",
            "test_sample_S1_L001_2.fastp.fastq.gz:md5,1c60f1f3958b8b2b0255b9fae6df0b3d",
            "test_sample_S1_L001_2.fastp.fastq.gz.md5:md5,01d742cc56639c476b0bac0c094eba4b",
            "test_sample_S1_L001_I1_001.fastp.fastq.gz.md5:md5,82e397b4a2418f27269a30f6ca0e349e",
            "test_sample_S1_L001_I1_001.fastq.gz:md5,018b0dbd8daf76fb9f6dfa4729ea0453",
            "test_sample_S1_L001_R1_001.fastq.gz:md5,df3ddf85d83bd722219c646c7815dbdc",
            "test_sample_S1_L001_R2_001.fastq.gz:md5,c8a387b517ff5b7e56b85ccbf882efd2",
            "Undetermined_S0_L001_I1_001.fastq.gz:md5,c93946d67bdb1d5b2ab955fd1dc9cb31",
            "Undetermined_S0_L001_R1_001.fastq.gz:md5,faa79ce215ae72fdc3375cda066c6757",
            "Undetermined_S0_L001_R2_001.fastq.gz:md5,d8955db4dfbf7aee89b6bdc3b7629f8f",
            true,
            true,
            "test_sample_S1_L001.fastp.json:md5,f616058446c5822797048ea6c5d2393d",
            "test_sample_S1_L001_I1_001.fastp.json:md5,42cad4ed99a3d577f100017cbe11d711",
            "IndexMetricsOut.bin:md5,e2ac28695537336b53a509f48e93ed1f"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-08-15T17:54:45.551098"
    }
}
